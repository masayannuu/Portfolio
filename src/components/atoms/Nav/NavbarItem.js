import React from 'react'
import styled from 'styled-components'

const NavAnker = styled.a`
  color: #ffffff;
  font-feature-settings: 'palt';
  letter-spacing: 0.2em;
  font-size: 81.25%;
  &:hover {
    color: #f0f0f0;
    text-decoration: underline;
    -webkit-transition: all 0.5s ease;
    -moz-transition: all 1s ease;
    text-shadow: #057D9F 0px 0px 20px;
  }
`

const NavbarItem = ({ children, href }) => (
  <div className="navbar-item">
    <p className="control">
      <NavAnker href={href}>{children}</NavAnker>
    </p>
  </div>
)

export default NavbarItem
