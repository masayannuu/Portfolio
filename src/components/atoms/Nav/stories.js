import React from 'react'
import { storiesOf } from '@storybook/react'

import NavbarItem from './NavbarItem'

storiesOf('Atoms.Nav', module).add('NavbarItem(white)', () => <NavbarItem hre="#">test</NavbarItem>)
