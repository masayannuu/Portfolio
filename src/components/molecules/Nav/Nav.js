import React from 'react'
import styled from 'styled-components'

import NavbarItem from '../../atoms/Nav/NavbarItem'

const NavMain = styled.nav`
  background-color: #a2c4c7;
`

const NavTitle = styled.a`
  font-size: 1rem;
  color: #ffffff;
  font-feature-settings: 'palt';
  letter-spacing: 0.2em;
  &:hover {
    color: #ffffff;
    text-decoration: underline;

`
// TODO: ハンバーガーメニューのアクティブロジック
const Nav = () => (
  <NavMain className="navbar" aria-label="main navigation">
    <div className="navbar-brand">
      <div className="navbar-item">
        <p className="control">
          <NavTitle href="/">Masaya Morimoto</NavTitle>
        </p>
      </div>
      <div role="button" className="navbar-burger" aria-label="menu" aria-expanded="false">
        <span aria-hidden="true" />
        <span aria-hidden="true" />
        <span aria-hidden="true" />
        <span aria-hidden="true" />
        <span aria-hidden="true" />
      </div>
    </div>
    <div className="navbar-menu">
      <div className="navbar-end">
        <NavbarItem href="#portfolio">PROFILE</NavbarItem>
        <NavbarItem href="#work">WORK</NavbarItem>
        <NavbarItem href="#sns">SNS</NavbarItem>
        <NavbarItem href="#blog">BLOG</NavbarItem>
        <NavbarItem href="#content">CONTACT</NavbarItem>
      </div>
    </div>
  </NavMain>
)

export default Nav
