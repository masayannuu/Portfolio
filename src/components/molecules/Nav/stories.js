import React from 'react'
import { storiesOf } from '@storybook/react'
import Nav from './Nav'

storiesOf('Molecules.Nav', module).add('Nav', () => <Nav />)
