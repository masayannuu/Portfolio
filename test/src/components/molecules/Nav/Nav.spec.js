import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Nav from '../../../../../src/components/molecules/Nav/Nav'

describe('Nav Nav Molecules', () => {
  it('should render Nav', () => {
    const wrapper = Enzyme.shallow(<Nav />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
