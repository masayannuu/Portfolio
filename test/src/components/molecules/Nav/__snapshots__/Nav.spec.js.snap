// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav Molecules should render Nav 1`] = `
<Nav__NavMain
  aria-label="main navigation"
  className="navbar is-fixed-top"
>
  <div
    className="navbar-brand"
  >
    <div
      className="navbar-item"
    >
      <Nav__NavTitle
        href="/"
      >
        MASAYA MORIMOTO
      </Nav__NavTitle>
    </div>
    <div
      aria-expanded={false}
      aria-label="menu"
      className="navbar-burger "
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </div>
  </div>
  <Nav__NavBar
    className="navbar-menu "
  >
    <div
      className="navbar-end"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <NavbarItem
        href="#profile"
      >
        PROFILE
      </NavbarItem>
      <NavbarItem
        href="#work"
      >
        WORK
      </NavbarItem>
      <NavbarItem
        href="#sns"
      >
        SNS
      </NavbarItem>
      <NavbarItem
        href="#contact"
      >
        SAY HELLO
      </NavbarItem>
    </div>
  </Nav__NavBar>
</Nav__NavMain>
`;
